notes-20200204.tex

```diff
@@ -0,0 +1,236 @@
+  \end{prooftree*}
+\item Elimination
+  \begin{mathpar}
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{\displaystyle\prod_{x{:}A}B}}
+        \hypo{\oftype{N}{A}}
+        \infer2{\oftype{\app{M}{N}}{B\subst{N}{x}}}
+      \end{prooftree*}
+    \end{varwidth}
+  \end{mathpar}
+\item Computation
+  \begin{mathpar}
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{x{:}A}{M}{B}}
+        \hypo{\oftype{N}{A}}
+        \infer2{\eqterm{\app{\parens{\lam{x}{M}}}{N}}{M\subst{N}{x}}{B\subst{N}{x}}}
+      \end{prooftree*}
+    \end{varwidth}
+  \end{mathpar}
+\item Uniqueness
+  \begin{mathpar}
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{\displaystyle\prod_{x{:}A}B}}
+        \infer1{\eqterm{M}{\lam{x}{\app{M}{x}}}{\displaystyle\prod_{x{:}A}B}}
+      \end{prooftree*}
+    \end{varwidth}
+  \end{mathpar}
+\end{enumerate}
+
+\section{Dependent Pair Types}
+\begin{enumerate}
+\item Formation:
+  \begin{prooftree*}
+    \hypo{\istype{A}}
+    \hypo{\istype{x{:}A}{B}}
+    \infer2{\istype{\dpairtype{x}{A}{B}}}
+  \end{prooftree*}
+\item Introduction
+  \begin{prooftree*}
+    \hypo{\oftype{M}{A}}
+    \hypo{\istype{x{:}A}{B}}
+    \hypo{\oftype{N}{B\subst{M}{x}}}
+    \infer3{\oftype{\pair{M}{N}}{\dpairtype{x}{A}{B}}}
+  \end{prooftree*}
+\item Elimination
+  \begin{mathpar}
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{\dpairtype{x}{A}{B}}}
+        \infer1{\oftype{\pairfst{M}}{A}}
+      \end{prooftree*}
+    \end{varwidth}
+    \and
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{\dpairtype{x}{A}{B}}}
+        \infer1{\oftype{\pairsnd{M}}{B\subst{\pairfst{M}}{x}}}
+      \end{prooftree*}
+    \end{varwidth}
+  \end{mathpar}
+  The second elimination rule depends on the first one.
+\item Computation
+  \begin{mathpar}
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{A}}
+        \hypo{\istype{x{:}A}{B}}
+        \hypo{\oftype{N}{B\subst{M}{x}}}
+        \infer3{\eqterm{\pairfst{\pair{M}{N}}}{M}{A}}
+      \end{prooftree*}
+    \end{varwidth}
+    \and
+    \begin{varwidth}{\textwidth}
+      \begin{prooftree*}
+        \hypo{\oftype{M}{A}}
+        \hypo{\istype{x{:}A}{B}}
+        \hypo{\oftype{N}{B\subst{M}{x}}}
+        \infer3{\eqterm{\pairsnd{\pair{M}{N}}}{N}{B\subst{M}{x}}}
+      \end{prooftree*}
+    \end{varwidth}
+  \end{mathpar}
+\item Uniqueness
+  \begin{prooftree*}
+    \hypo{\oftype{M}{\dpairtype{x}{A}{B}}}
+    \infer1{\eqterm{\pair{\pairfst{M}}{\pairsnd{M}}}{M}{\dpairtype{x}{A}{B}}}
+  \end{prooftree*}
+  Notice that the elimination rules tell us
+  $\oftype{\pairfst{M}}{A}$ and $\oftype{\pairsnd{M}}{B\subst{\pairfst{M}}{x}}$.
+  Then by the introduction rule we know
+  $\oftype{\pair{\pairfst{M}}{\pairsnd{M}}}{\dpairtype{x}{A}{B}}$.
+\end{enumerate}
+
+\section{Dependent Sum Types}
```

Comment:
I probably made a mistake. This should be "disjoint sum types".

Reply:
Oops it's my fault. I added "disjoint" in the first section but forgot to change this one.